src/commands/environments/get.ts

```diff
@@ -21,34 +21,34 @@ export default class DetailedEnvironments extends Base {
 
     public async run(): Promise<void> {
         const { flags } = await this.parse(DetailedEnvironments)
-        const keys = flags['keys']?.split(',') || []
+        const keys = flags['keys']?.split(',')
         const { headless, project } = flags
         await this.requireProject(project, headless)
 
-        let environments = await fetchEnvironments(this.authToken, this.projectKey)
-        if (keys.length) {
-            environments = environments.filter((environment) => 
-                keys.includes(environment.key) || keys.includes(environment._id))
+        if (keys) {
+            const environments = await Promise.all(
```

Comment:
Since a user could pass any number of keys, we should create batches of requests. You could checkout [lodash's chunk method](https://lodash.com/docs/4.17.15#chunk) for splitting the array into groups of requests

Reply:
oh i see, would the idea be to prevent sending too many requests in parallel?

Reply:
Yeah exactly, that way we can choose a reasonable number of requests to send in parallel (say ~10), and then even if the user passes 100+ keys we can await each batch rather than spamming our API

Reply:
makes sense. i added that plus I added `.filter(Boolean)` to the results to avoid null/ undefined in the results for keys that aren't found